Mathlib/LinearAlgebra/FiniteDimensional/Basic.lean

```diff
@@ -393,6 +393,26 @@ theorem ofInjectiveEndo_left_inv (f : V →ₗ[K] V) (h_inj : Injective f) :
     ((ofInjectiveEndo f h_inj).symm : V →ₗ[K] V) * f = 1 :=
   LinearMap.ext <| (ofInjectiveEndo f h_inj).symm_apply_apply
 
+variable {V' : Type*} [AddCommGroup V'] [Module K V'] [FiniteDimensional K V']
+omit [FiniteDimensional K V]
+
+/-- An injective linear map between finite-dimensional modules of equal rank
+is a linear equivalence.
+
+Unlike `LinearEquiv.ofFinrankEq` (which creates an *abstract* linear equivalence from `V` to `V'`),
+this lemma improves a *given* injective linear map to a linear equivalence.
+-/
+noncomputable def ofInjectiveOfFinrankEq (f : V →ₗ[K] V') (hinj : Function.Injective f)
+    (hrank : Module.finrank K V = Module.finrank K V') : V ≃ₗ[K] V' :=
+  haveI : LinearMap.range f = ⊤ :=
+    Submodule.eq_top_of_finrank_eq ((LinearMap.finrank_range_of_inj hinj).trans hrank)
+  (ofInjective f hinj).trans (ofTop (LinearMap.range f) this)
+
+@[simp]
+lemma ofInjectiveOfFinrankEq_coe (f : V →ₗ[K] V') (hinj : Function.Injective f)
```

Comment:
Shouldn't this be called `coe_ofInjectiveOfFinrankEq`? (based on `coe_ofInjectiveEndo` above)

Reply:
Good point, thanks!
```suggestion
lemma coe_ofInjectiveOfFinrankEq (f : V →ₗ[K] V') (hinj : Function.Injective f)